pkg/prometheus/promcfg.go

```diff
@@ -740,112 +740,110 @@ func (cg *configGenerator) generateProbeConfig(
 		}...)
 
 		cfg = append(cfg, yaml.MapItem{Key: "relabel_configs", Value: enforceNamespaceLabel(relabelings, m.Namespace, enforcedNamespaceLabel)})
+		return cfg
```

Comment:
It would mean that if `StaticConfig` and `Ingress` are defined at the same time, only the former would be used. I'm not sure this is what we wanted initially. @lilic?

Reply:
From what i can see, i didn't change the behavior, as the two if statements for checking the content of m.Spec.Targets.StaticConfig are complementary. If this is no intentional behavior, i can revert this and we can file an issue for it

Reply:
@sschne totally agree. I was pinging @lilic for clarifications so we don't forget :)

Reply:
done